src/map/map.c

```diff
@@ -5516,6 +5552,46 @@ void read_map_zone_db(void) {
 				zone->disabled_skills_count = disabled_skills_count;
 			}
 
+			if ((statuses = libconfig->setting_get_member(zone_e, "disabled_status")) != NULL) {
+				disabled_status_count = libconfig->setting_length(statuses);
+				/* validate */
+				for(h = 0; h < libconfig->setting_length(statuses); h++) {
+					struct config_setting_t *status_info = libconfig->setting_get_elem(statuses, h);
+					int status_value = SC_NONE;
+					name = config_setting_name(status_info);
+					if (!script->get_constant(name, &status_value) || status_value <= SC_NONE || status_value >= SC_MAX) {
+						ShowError("map_zone_db: unknown status (%s) in disabled_status for zone '%s', skipping skill...\n", name, zone->name);
+						libconfig->setting_remove_elem(statuses, h);
+						--disabled_status_count;
+						--h;
+						continue;
+					}
+
+					if( !libconfig->setting_get_bool(status_info) )/* we don't remove it from the three due to inheritance */
+						--disabled_status_count;
+				}
+				/* all ok, process */
+				CREATE(zone->disabled_status, int, disabled_status_count);
```

Comment:
This will wastefully execute a few statements, none of which check for the number of elements to create since `disabled_status_count` here can be 0, causing a slight performance degradation. (Wonder why there are no `<= 0` checks within the macros.) Either way, anything depending on `disabled_status_count` should be within a scope that guarantees it being greater than 0.

Reply:
It's same as all above checks (for other features such as disabled_skills,disabled_items, etc)